process/src/SpriteProcessor.js

```diff
@@ -0,0 +1,142 @@
+"use strict";
+
+const Canvas = require('canvas');
+const fs = require('fs');
+
+class SpriteProcessor {
+  constructor(spritesDir, pngDir) {
+    this.spritesDir = spritesDir;
+    this.pngDir = pngDir;
+    this.canvas = new Canvas(512, 1024);
+    this.context = this.canvas.getContext('2d');
+  }
+
+  process(objects) {
+    for (var id in objects) {
+      this.processObject(objects[id]);
+    }
+  }
+
+  processObject(object) {
+    this.context.setTransform(1, 0, 0, 1, 0, 0);
+    this.context.clearRect(0, 0, this.canvas.width, this.canvas.height);
+
+    for (var sprite of object.sprites) {
+      this.parseSpriteFile(sprite);
+      this.drawSprite(sprite);
+    }
+
+    const bounds = this.objectBounds(object);
+    const width = bounds.maxX - bounds.minX;
+    const height = bounds.maxY - bounds.minY;
+
+    const newCanvas = new Canvas(Math.max(width, 128), Math.max(height, 128));
+    const newContext = newCanvas.getContext('2d');
+
+    newContext.setTransform(1, 0, 0, 1, 0, 0);
+
+    newContext.drawImage(
+      this.canvas,
+      bounds.minX + this.canvas.width / 2,
+      -bounds.maxY + this.canvas.height / 2,
+      width,
+      height,
+      (newCanvas.width - width) / 2,
+      (newCanvas.height - height) / 2,
+      width,
+      height
+    );
+
+    // Debug
+    this.context.beginPath(); //
```

Comment:
Should this still be in here?

Reply:
I left it in just in case. Feel free to remove.